system/jlib/jfile.cpp

```diff
@@ -7870,14 +7903,34 @@ MODULE_INIT(INIT_PRIORITY_STANDARD)
             values[BlockedSequentialIO] = plane.getPropInt(("@" + std::string(planeAttributeTypeStrings[BlockedSequentialIO])).c_str()) * 1024;
             values[BlockedRandomIO] = plane.getPropInt(("@" + std::string(planeAttributeTypeStrings[BlockedRandomIO])).c_str()) * 1024;
         }
+
+        // NB: will not pick up bare-metal dafilesrv config because it does not have a component.xml
+        bool hasFlags = getComponentConfigSP()->hasProp("expert/@fileFlagsMask");
+        if (hasFlags)
+        {
+            StringBuffer expertFlagsStr;
+            getComponentConfigSP()->getProp("expert/@fileFlagsMask", expertFlagsStr);
+            expertFileFlagsMask = getFlagsFromStr(hasExpertFileFlags, expertFlagsStr);
+        }
+        else
+        {
+            hasFlags = getGlobalConfigSP()->hasProp("expert/@fileFlagsMask");
+            if (hasFlags)
+            {
+                StringBuffer expertFlagsStr;
+                getGlobalConfigSP()->getProp("expert/@fileFlagsMask", expertFlagsStr);
+                expertFileFlagsMask = getFlagsFromStr(hasExpertFileFlags, expertFlagsStr);
+            }
+        }
```

Comment:
suggestion: have separate enable and disable options (so some flags could be enabled and other disabled), and I think the code is simpler then, e.g:
```
        // defaults
        expertEnableIFileFlagsMask = IFEnone;
        expertDisableIFileFlagsMask = IFEnone;

        // leave for compatibility
        if (!isPCFlushAllowed())
            expertDisableIFileFlagsMask = IFEnocache;

        StringBuffer fileFlagsStr;
        if (getComponentConfigSP()->getProp("expert/@enableIFileMask", fileFlagsStr) ||
            getGlobalConfigSP()->getProp("expert/@enableIFileMask", fileFlagsStr))
        {
            expertEnableIFileFlagsMask = static_cast<IFEflags>(expertEnableIFileFlagsMask | strtoul(fileFlagsStr, NULL, 0));
        }
        if (getComponentConfigSP()->getProp("expert/@disableIFileMask", fileFlagsStr.clear()) ||
            getGlobalConfigSP()->getProp("expert/@disableIFileMask", fileFlagsStr))
        {
            expertDisableIFileFlagsMask = static_cast<IFEflags>(expertDisableIFileFlagsMask | strtoul(fileFlagsStr, NULL, 0));
        }
```